### DISCUSSION TITLE
How do I remove a method parameter?

### DISCUSSION THREAD

**Discussion by grosch-intl** [eslint/eslint#17044] [Q&A]
When I drill into my MethodDefinition and then FunctionExpression and I get the parameter that I want to remove from the method declaration, how do I actually do that?  If I just call a fixer to removeNode, that'll leave corrupt code if there are multiple items.  So I'd go from:

```ts
constructor(first: Something, second: SomethingElse)
```

to

```ts
constructor(first: Something, )
```

So I thought I'd do something like this:

```ts
const source = context.getSourceCode()
const tokenBefore = source.getTokenBefore(node)
const hasCommaBefore = isPunctuatorToken(tokenBefore) && tokenBefore.value === ','

const tokenAfter = source.getTokenAfter(node)
const hasCommaAfter = isPunctuatorToken(tokenAfter) && tokenAfter.value === ','
```

and then modify the range, but then I run into issues of it crossing multiple lines.  Am I supposed to do it like so:

```ts
context.report({
    messageId: 'injectItem',
    node,
    fix: fixer => {
        const rules: TSESLint.RuleFix[] = [ fixer.remove(node) ]
     
        if (hasCommaBefore && tokenBefore)
            rules.push(fixer.remove(tokenBefore))
        else if (hasCommaAfter && tokenAfter)
            rules.push(fixer.remove(tokenAfter))
              
        return rules
    }
})
```

**Comment by mdjermanovic** [comment]
> and then modify the range, but then I run into issues of it crossing multiple lines

Can you please clarify this, what are the issues with crossing multiple lines?
